pkg/providers/azure_test.go

```diff
@@ -307,3 +328,73 @@ func TestAssignSpRole(t *testing.T) {
 		})
 	}
 }
+
+func TestCreateAzApp(t *testing.T) {
```

Comment:
You could probably merge the tests for createAzApp into table-driven test as well

Reply:
yeah, will do 👍